dockerfiles/ci/buster/php-8.1/Dockerfile

```diff
@@ -0,0 +1,118 @@
+        --with-config-file-scan-dir=${PHP_INSTALL_DIR_DEBUG_NTS}/conf.d; \
+    make -j "$((`nproc`+1))"; \
+    make install; \
+    mkdir -p ${PHP_INSTALL_DIR_DEBUG_NTS}/conf.d;
+
+FROM build as php-nts
+RUN set -eux; \
+    mkdir -p /tmp/build-php && cd /tmp/build-php; \
+    /home/circleci/configure.sh \
+        --prefix=${PHP_INSTALL_DIR_NTS} \
+        --with-config-file-path=${PHP_INSTALL_DIR_NTS} \
+        --with-config-file-scan-dir=${PHP_INSTALL_DIR_NTS}/conf.d; \
+    make -j "$((`nproc`+1))"; \
+    make install; \
+    mkdir -p ${PHP_INSTALL_DIR_NTS}/conf.d;
+
+FROM build as php-zts
+RUN set -eux; \
+    mkdir -p /tmp/build-php && cd /tmp/build-php; \
+    /home/circleci/configure.sh \
+        --enable-zts \
+        --prefix=${PHP_INSTALL_DIR_ZTS} \
+        --with-config-file-path=${PHP_INSTALL_DIR_ZTS} \
+        --with-config-file-scan-dir=${PHP_INSTALL_DIR_ZTS}/conf.d; \
+    make -j "$((`nproc`+1))"; \
+    make install; \
+    mkdir -p ${PHP_INSTALL_DIR_ZTS}/conf.d;
+
+FROM base as final
+COPY --chown=circleci:circleci --from=build $PHP_SRC_DIR $PHP_SRC_DIR
+COPY --chown=circleci:circleci --from=php-debug-zts-asan $PHP_INSTALL_DIR_DEBUG_ZTS_ASAN $PHP_INSTALL_DIR_DEBUG_ZTS_ASAN
+COPY --chown=circleci:circleci --from=php-debug $PHP_INSTALL_DIR_DEBUG_NTS $PHP_INSTALL_DIR_DEBUG_NTS
+COPY --chown=circleci:circleci --from=php-nts $PHP_INSTALL_DIR_NTS $PHP_INSTALL_DIR_NTS
+COPY --chown=circleci:circleci --from=php-zts $PHP_INSTALL_DIR_ZTS $PHP_INSTALL_DIR_ZTS
+
+RUN set -eux; \
+    for phpVer in $(ls ${PHP_INSTALL_DIR}); \
+    do \
+        echo "Install exts for PHP $phpVer..."; \
+        switch-php ${phpVer}; \
+        pecl channel-update pecl.php.net; \
+        iniDir=$(php -i | awk -F"=> " '/Scan this dir for additional .ini files/ {print $2}'); \
+        \
+        curl https://github.com/krakjoe/apcu/archive/refs/heads/master.zip -LO; \
+        unzip master.zip; \
+        cd apcu-master; \
+        phpize; \
+        ./configure; \
+        make -j "$((`nproc`+1))" install; \
+        echo "extension=apcu.so" >> ${iniDir}/apcu.ini; \
+        pecl install ast; echo "extension=ast.so" >> ${iniDir}/ast.ini; \
+        #yes 'no' | pecl install memcached; \ # TODO Does not support PHP 8 yet
+        # pecl install mongodb; echo "extension=mongodb.so" >> ${iniDir}/mongodb.ini; \ # TODO Does not support PHP 8.1 yet
+        pecl install redis; echo "extension=redis.so" >> ${iniDir}/redis.ini; \
+        # Xdebug is disabled by default \
+        curl https://github.com/xdebug/xdebug/archive/refs/heads/master.zip -LO; \
+        unzip master.zip; \
+        cd xdebug-master; \
+        phpize; \
+        ./configure; \
+        make -j "$((`nproc`+1))" install; \
+        cd $(php-config --extension-dir); \
+        mv xdebug.so xdebug-3.1.0.so; \
+    done;
+
+RUN set -eux; \
+# Set the default PHP version
+    switch-php debug;
+
+# Install Composer
+COPY --from=composer:1 /usr/bin/composer /usr/local/bin/composer
```

Comment:
Not `composer:2`? We started migrating to 2 where we can (though some jobs were having trouble).

Reply:
No idea, I think we shouldn't open that can of worms immediately though, but in a separate PR.